utils/filecache/entrymap.go

```diff
@@ -0,0 +1,42 @@
+package filecache
+
+import "sync"
+
+type entryMap struct {
+	entries sync.Map
```

Comment:
can you justify the use of sync.Map vs a map with a mutex.

Since sync.Map is not recommended to be used unless there are specific conditions, I think it needs a comment explaining why it is used.

See https://pkg.go.dev/sync#Map